### PR TITLE
agent: add support to FQDN in ManagerHost

### PR DESCRIPTION
**Pull request by dougsland**
GitHub-Issue: https://github.com/containers/hirte/issues/276

### CODE REVIEW

File: src/libhirte/test/common/network/get_address_test.c [hunk 1]
```diff
@@ -0,0 +1,48 @@
+/* SPDX-License-Identifier: GPL-2.0-or-later */
+#include <assert.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "libhirte/common/network.h"
+
+#define bool_to_str(b) b ? "true" : "false"
```
Comment (engelmi) [inline src/libhirte/test/common/network/get_address_test.c]: Nit: `bool_to_str` was only used in `string-utils_test` before. We could move it now to a central location, maybe [libhirte/common/string-utils](https://github.com/containers/hirte/blob/main/src/libhirte/common/string-util.h) or [libhirte/common/common)(https://github.com/containers/hirte/blob/main/src/libhirte/common/common.h)?

File: src/libhirte/common/network.c [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        if (status != 0) {
+                fprintf(stderr, "getaddrinfo failed: %s\n", gai_strerror(status));
+                return 1;
+        }
+
+        void *addr = NULL;
+        if (res->ai_family == AF_INET) { // IPv4 address
+                struct sockaddr_in *ipv4 = (struct sockaddr_in *) res->ai_addr;
+                addr = &(ipv4->sin_addr);
+                if (inet_ntop(AF_INET, addr, ip_address, ip_address_size) == NULL) {
```
Comment (engelmi) [inline src/libhirte/common/network.c]: Nit: Could be replaced with the `typesafe_inet_ntop4`.

File: src/libhirte/common/network.c [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        }
+
+        struct addrinfo hints, *res = NULL;
+        memset(&hints, 0, sizeof(hints));
+        hints.ai_family = AF_UNSPEC;
+        hints.ai_socktype = SOCK_STREAM;
+
+        int status = getaddrinfo(domain, NULL, &hints, &res);
+        if (status != 0) {
+                fprintf(stderr, "getaddrinfo failed: %s\n", gai_strerror(status));
```
Comment (engelmi) [inline src/libhirte/common/network.c]: Nit: This and the following `fprint`s can be replaced by `hirte_log_errorf`.

**Review by engelmi — COMMENTED** [review]
Looks great! I only have a few nit-picky comments. 

And thanks for adding unit tests! Having one test file for one function under test seems better than one test file containing multiple (as it is currently - we'll have to refactor that).

**Comment by dougsland** [comment]
@engelmi ouch not sure how to access the logs from GitHub actions:
```
>>> MALLOC_PERTURB_=199 valgrind --leak-check=full --error-exitcode=1 --track-origins=yes /__w/hirte/hirte/builddir/src/libhirte/test/common/network/get_address_test
```

Maybe setup a valgrind env here?

```
[root@node1 hirte]# meson test -C builddir
ninja: Entering directory `/root/hirte/builddir'
ninja: no work to do.
1/8 cfg_test                    OK              0.03s
2/8 list_test                   OK              0.03s
3/8 parse-util_test             OK              0.03s
4/8 string-util_test            OK              0.03s
5/8 isIPv4Addr_test             OK              0.02s
6/8 isIPv6Addr_test             OK              0.02s
7/8 assemble_tcp_address        OK              0.01s
8/8 get_address_test            OK              0.02s

Ok:                 8
Expected Fail:      0
Fail:               0
Unexpected Pass:    0
Skipped:            0
Timeout:            0

Full log written to /root/hirte/builddir/meson-logs/testlog.txt
```

**Comment by mwperina** [comment]
> @engelmi ouch not sure how to access the logs from GitHub actions:

Maybe https://github.com/containers/hirte/pull/346 will help here

**Comment by pypingou** [comment]
Could you rebase to see if the new CI config helps?

**Comment by dougsland** [comment]
> > @engelmi ouch not sure how to access the logs from GitHub actions:
> 
> Maybe #346 will help here

cool, let me try.

**Comment by dougsland** [comment]
do we have tools commands like /retest? @pypingou @mwperina 

/retest

**Comment by dougsland** [comment]
@mwperina @engelmi @rhatdan @pypingou (and others) ready for review and fly.

**Comment by engelmi** [comment]
> do we have tools commands like /retest? @pypingou @mwperina
> 
> /retest

No, at least there is no bot configured that would pick up those commands. We are using "plain" github actions. So if a step fails, its best to click on "Details" - on the following page you can trigger a re-run manually, look at the CI logs and download artifacts (like the unit test logs).

**Review by engelmi — APPROVED** [review]
LGTM

**Pull request merged**